src/main/java/org/zalando/fahrschein/NakadiReader.java

```diff
@@ -121,9 +122,20 @@ private Cursor readCursor(JsonParser jsonParser) throws IOException {
 
         final List<T> events = new ArrayList<>();
 
-        while (jsonParser.nextToken() == JsonToken.START_OBJECT) {
-            final T event = objectReader.readValue(jsonParser, eventClass);
-            events.add(event);
+        if (jsonParser.nextToken() == JsonToken.START_OBJECT) {
+            final Iterator<T> eventIterator = objectReader.readValues(jsonParser, eventClass);
+            while (eventIterator.hasNext()) {
+                try {
+                    events.add(eventIterator.next());
+                } catch (final Exception e) {
```

Comment:
This could be more explicit by only catching `RuntimeException` or even `RuntimeJsonMappingException`. There is another `onError` method in `CursorManager` so maybe it makes sense for this to only handle mapping exceptions.